backend/tetra/data/models/result.py

```diff
@@ -191,3 +191,27 @@ def _get_results_metadata(cls, and_clause, handler=None):
         count_results = handler.get_all(resource_class=Result, query=query)
 
         return ResultMetadata.from_database_counts(count_results)
+
+    @classmethod
+    def get_test_stats(self, since_time=0, handler=None):
+        """Returns the spurious failures in our nightlies for the given time
+        interval [time_since, now]
+
+        """
+
+        handler = handler or get_handler()
+
+        from sqlalchemy.sql import text
+        statement = text(
+            "SELECT test_name, count(result) "
+                "FROM (SELECT name, timestamp, test_name, result FROM results JOIN builds ON results.build_id = builds.id) as tmp "
+                "WHERE result = 'failed' "
+                "OR result = 'error' "
+                "AND name like 'nightly%' "
+                "AND timestamp > :since_time "
+                "GROUP BY test_name "
+                "ORDER BY count(result) DESC"
+        )
```

Comment:
in general it's ok, but plain SQL strings like this tend to be a bit more brittle than using the ORM functionality (especially when it comes to working with a changing db)... OTOH it might be harder to figure out a good way to reach that same query...

still, looking at the rest of the file: consistency... 😛

Reply:
I did dig deep into the `alchemy` docs, and did manage to cook together something which works actually.

It was not a quick endeavor! 

:p